src/generators/legacy-json/utils/buildHierarchy.mjs

```diff
@@ -0,0 +1,63 @@
+/**
+ * We need the files to be in a hierarchy based off of depth, but they're
+ *  given to us flattened. So, let's fix that.
+ *
+ * Assuming that {@link entries} is in the same order as the elements are in
+ *  the markdown, we can use the entry's depth property to reassemble the
+ *  hierarchy.
+ *
+ * If depth <= 1, it's a top-level element (aka a root).
+ *
+ * If it's depth is greater than the previous entry's depth, it's a child of
+ *  the previous entry. Otherwise (if it's less than or equal to the previous
+ *  entry's depth), we need to find the entry that it was the greater than. We
+ *  can do this by just looping through entries in reverse starting at the
+ *  current index - 1.
+ *
+ * @param {Array<ApiDocMetadataEntry>} entries
+ * @returns {Array<import('../types.d.ts').HierarchizedEntry>}
+ */
+export function buildHierarchy(entries) {
+  const roots = [];
+
+  for (let i = 0; i < entries.length; i++) {
+    const entry = entries[i];
+    const currentDepth = entry.heading.depth;
+
+    if (currentDepth <= 1) {
+      // We're a top-level entry
+      roots.push(entry);
+      continue;
+    }
+
+    const previousEntry = entries[i - 1];
+
+    const previousDepth = previousEntry.heading.depth;
+    if (currentDepth > previousDepth) {
```

Comment:
Can we add some blank lines and some inline docs with some more explanation about what the code is doing and why also I < 2?